### PR TITLE
chore(tracing): switch all db tags involving row counts to db.row_count

### PR DESCRIPTION
**Pull request by wconti27**
## Description
Switches `db.rowcount`, `sql.rows` and any other previous database row count tag to only `db.row_count` to match new Datadog Span specification.

## Related PRs
This PR is broken up into several PR's to make it more manageable.

All other PR's are based on the `align-db.row_count-across-integrations` branch. This branch will still contain changes for updating the existing `sql.rows`, `db.rowcount`, and `cassandra.rows` tags to the new tag name. Any constants associated with old tags will also be deprecated with this PR.

Other PRs:
[- Implement `db.row_count` for caching integrations (no redis)](https://github.com/DataDog/dd-trace-py/pull/5052)
[- Implement `db.row_count` for Pymongo](https://github.com/DataDog/dd-trace-py/pull/5053)
[- Implement `db.row_count` for Redis and Redis-like integrations ](https://github.com/DataDog/dd-trace-py/pull/5054)

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Change contains telemetry where appropriate (logs, metrics, etc.).
- [x] Telemetry is meaningful, actionable and does not have the potential to leak sensitive data.

### CODE REVIEW

File: ddtrace/ext/sql.py [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 # tags
 QUERY = "sql.query"  # the query text
-ROWS = "sql.rows"  # number of rows returned by a query
```
Comment (brettlangdon) [inline ddtrace/ext/sql.py:6]: we cannot delete this, we have to deprecate it.

File: ddtrace/ext/cassandra.py [hunk 2]
```diff
@@ -3,5 +3,4 @@
 KEYSPACE = "cassandra.keyspace"
 CONSISTENCY_LEVEL = "cassandra.consistency_level"
 PAGINATED = "cassandra.paginated"
-ROW_COUNT = "cassandra.row_count"
```
Comment (brettlangdon) [inline ddtrace/ext/cassandra.py:6]: we cannot delete this, we have to deprecate it.

**Review by brettlangdon — DISMISSED** [review]
would you be willing to break this into multiple PRs?

if it was just updating existing span tag names it probably wouldn't be a big deal, but given that we are introducing new logic into some of the integrations we should break up.

you could group them, like all the redis changes together in one PR, etc.

**Review by emmettbutler — COMMENTED** [review]
Aside from @brettlangdon's comments I have nothing to add.

**Comment by wconti27** [comment]
@brettlangdon @emmettbutler I updated this branch and separated PR's as Brett suggested, should be easier to review now. This PR is solely for switching existing tags now. Will add the deprecations as well. Cheers

**Review by emmettbutler — CHANGES_REQUESTED** [review]
Looks good to me aside from the need to deprecate rather than delete the `ext` constants.

File: releasenotes/notes/deprecate-constants-5e6e2cfdb76662ba.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    ``ddtrace.ext.sql.ROWS`` is deprecated. Use ``ddtrace.ext.db.ROWCOUNT`` instead.
+  - |
+    ``ddtrace.ext.cassandra.ROW_COUNT`` is deprecated. Use ``ddtrace.ext.db.ROWCOUNT`` instead.
```
Comment (mabdinur) [inline releasenotes/notes/deprecate-constants-5e6e2cfdb76662ba.yaml:6]:
```suggestion
  - |
    dbapi: ``ddtrace.ext.sql.ROWS`` is deprecated. Use ``ddtrace.ext.db.ROWCOUNT`` instead.
  - |
    cassandra: ``ddtrace.ext.cassandra.ROW_COUNT`` is deprecated. Use ``ddtrace.ext.db.ROWCOUNT`` instead.

```

**Review by mabdinur — COMMENTED** [review]
Aside from release notes LGTM. Thanks for pushing this through.

File: releasenotes/notes/deprecate-constants-5e6e2cfdb76662ba.yaml (hunk 3 above)
Comment (wconti27) [reply releasenotes/notes/deprecate-constants-5e6e2cfdb76662ba.yaml]: should be all set, updated the release notes

**Review by emmettbutler — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**